### PR TITLE
Refactor for null checking on AvatarMovementController

### PR DESCRIPTION
**Pull request by PalauiDCL**
The every frame null checking for every avatar has now converted to a boolean checking increasing performance for this class.
I also made the code style more consistent.

For testers there should not be anything to test a part that the characters keep behaving exactly the same, no modifications should exist at all.

### CODE REVIEW

**Comment by PalauiDCL** [comment]
Linked issue: https://app.zenhub.com/workspaces/explorer-6047bad476c3c0001942ee91/issues/decentraland/unity-renderer/2715

**Review by AjimenezDCL — APPROVED** [review]
It looks good but I'd like to have some benchmarks on the benefits. Approving regardless

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMovementController.cs [hunk 1]
```diff
@@ -7,66 +7,84 @@ namespace DCL
...
+
+        private Transform avatarTransformValue;
+
+        private Quaternion targetRotation;
+        private Vector3 currentWorldPosition = Vector3.zero;
+        private Vector3 targetPosition;
 
         private float movementLerpWait = 0f;
         private float movementLerpWaitCounter = 0f;
+        private bool avatarTransformAvailable = false;
```
[the remark this answers is not in the document]
Comment (PalauiDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMovementController.cs:26]: Not yet, will bring it before merge

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMovementController.cs [hunk 2]
```diff
@@ -7,66 +7,84 @@ namespace DCL
...
+
 
-        Transform avatarTransform
+        private Transform AvatarTransform
```
[the remark this answers is not in the document]
Comment (PalauiDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Avatar/AvatarMovementController.cs:29]: My mistake, wanted to convert it into a Property and changed the idea after the renaming, correcting it back

**Pull request merged**